src/Converter_TEST.cc

```diff
@@ -2141,6 +2141,364 @@ TEST(Converter, Pose_16_to_17)
   EXPECT_STREQ("parent", jointLinkPoseElem->Attribute("relative_to"));
 }
 
+const std::string CONVERT_DOC_17_18 =
+  sdf::filesystem::append(PROJECT_SOURCE_PATH, "sdf", "1.8", "1_7.convert");
+
+/////////////////////////////////////////////////
+/// Test conversion unflattened world in 1.7 to 1.8
+TEST(Converter, World_17_to_18)
+{
+  // for ElementToString
+  using namespace sdf::SDF_VERSION_NAMESPACE;
+
+  // ------- The flattened world in 1.7 format
+  std::string xmlString = R"(
+  <?xml version="1.0" ?>
+  <sdf version='1.7'>
+    <world name="default">
+      <model name='include_links'>
+        <frame name='A::__model__' attached_to='A::B::C'>
+          <pose relative_to='__model__'>1 0 0 0 0 0</pose>
+        </frame>
+        <frame name='A::B::__model__' attached_to='A::B::C'>
+          <pose relative_to='A::__model__'>0 1 0 0 0 0</pose>
+        </frame>
+        <link name='A::B::C'>
+          <pose relative_to='A::B::__model__'>0 0 1 0 0 0</pose>
+        </link>
+      </model>
+    </world>
+  </sdf>)";
+
+
+  tinyxml2::XMLDocument xmlDoc;
+  xmlDoc.Parse(xmlString.c_str());
+
+  // Convert
+  tinyxml2::XMLDocument convertXmlDoc;
+  convertXmlDoc.LoadFile(CONVERT_DOC_17_18.c_str());
+  sdf::Converter::Convert(&xmlDoc, &convertXmlDoc);
+
+  // Compare converted xml with expected
+  std::string convertedXmlStr = ElementToString(xmlDoc.RootElement());
+  ASSERT_FALSE(convertedXmlStr.empty());
+
+  std::string expectedXmlStr = R"(
+  <sdf version="1.7">
+      <world name="default">
+          <model name="include_links">
+              <model name="A" canonical_link="B::C">
+                  <pose relative_to="__model__">1 0 0 0 0 0</pose>
+                  <model name="B" canonical_link="C">
+                      <pose relative_to="__model__">0 1 0 0 0 0</pose>
+                      <link name="C">
+                          <pose relative_to="__model__">0 0 1 0 0 0</pose>
+                      </link>
+                  </model>
+              </model>
+          </model>
+      </world>
+  </sdf>)";
+
+  tinyxml2::XMLDocument expectedXmlDoc;
+  expectedXmlDoc.Parse(expectedXmlStr.c_str());
+
+  EXPECT_EQ(ElementToString(expectedXmlDoc.RootElement()), convertedXmlStr);
+
+    // Check some basic elements
+  tinyxml2::XMLElement *convertedElem =  xmlDoc.FirstChildElement();
```

Comment:
Can you add `ASSERT_NE(nullptr, convertElem)` whenever you get child elements?

Reply:
3e0cd5d70fce3d7dbe32746c2f3ce526ab059baa